scripts/csv-to-json/index.ts

```diff
@@ -22,49 +30,80 @@ function csvToType(rowStrings: string[]): any {
-    const row = rowString.split(',')
-    row.forEach((item, i) => {
-      const heading = firstRow[i]
-      if (!heading) throw new Error("No heading")
-      const keystring = row[0]
-      if (!keystring) throw new Error("No keystring in item")
-      if (i === 0) {
-        if (heading !== "keystring") throw new Error ("Expected 'keystring' heading")
-        const current = res[keystring]
-        if (!current) {
-          res[keystring] = [{}] as any[]
-        } else {
-          res[keystring] = current.concat([{}])
-        }
-      } else {
-        const objectArray: any[] = res[keystring]
-        if (!objectArray) throw new Error("No object array")
-        if (objectArray.length === 0) objectArray.push({})
-        let object = objectArray[objectArray.length - 1]
-        if (object === undefined) throw new Error("No object")
-        object[heading] = item
-        objectArray[objectArray.length - 1] = object
-        res[keystring] = objectArray
-      }
-    })
-  })
+  const type = firstRow[0]
+  let res: any = {}
+  switch (type) {
+    case undefined:
+      throw new Error("No type")
+
+    case "keystring":
+      rowStrings.forEach(rowString => {
+        const row = rowString.split(',')
+        row.forEach((item, i) => {
+          const heading = firstRow[i]
+          if (!heading) throw new Error("No heading")
+          const keystring = row[0]
+          if (!keystring) throw new Error("No keystring in item")
+          if (i === 0) {
+            const current = res[keystring]
+            if (!current) {
+              res[keystring] = [{}] as any[]
+            } else {
+              res[keystring] = current.concat([{}])
+            }
+          } else {
+            const objectArray: any[] = res[keystring]
+            if (!objectArray) throw new Error("No object array")
+            if (objectArray.length === 0) objectArray.push({})
+            let object = objectArray[objectArray.length - 1]
+            if (object === undefined) throw new Error("No object")
+            object[heading] = item
+            objectArray[objectArray.length - 1] = object
+            res[keystring] = objectArray
+          }
+        })
+      })
+      break
+
+    default:
+      // treat as array
+      if (typeof res === "object") res = [] as string[]
+      rowStrings.forEach(rowString => {
+        res.push(rowString)
+      })
+  }
   return res
 }
 
-function loadCsv(fileName: string): string[] {
-  const csv = fs.readFileSync(`${__dirname}/csv/${fileName}.csv`)
+function loadCsv(dirPath: string, fileName: string): string[] {
+  const csv = fs.readFileSync(`${dirPath}/${fileName}`)
   const array = csv.toString('utf8').split('\r\n')
-  const truncArray = array.map(string => {
-    const index = string.indexOf(',,,')
-    if (index >= 0) {
-      const row = string.slice(0, index)
-      if (!row) throw new Error("No row")
-      return row
+  const truncArray = []
+  for (let i = 0; i < array.length; i++) {
```

Comment:
Would prefer the modern construction `for (i in array) { const string = array[i] ... }`